py/src/nupic/optimization/ae.py

```diff
@@ -0,0 +1,898 @@
+#!/usr/bin/python3
+"""
+Automatic Experimenter
+Written by David McDougall, 2018-2019
+
+This is a parameter optimization framework.
+ * It methodically records the results of different sets of parameters and
+analyses the results.  It then automatically suggests and evaluates
+modifications to the parameters.
+ * It exposes a convenient API for users to hook their program into this.
+ * The framework allows for testing each set of parameters several times and
+calculates the average and standard deviation of the results.  It also
+calculates the confidence that a parameter change caused the score to change.
+ * It is extensible: new methods for automated parameter optimization can be
+added.  Currently this implements a basic grid search strategy.  In the future I
+hope to add a particle swarming method.
+
+To use this module, structure experiments as follows:
+    ExperimentModule is a python3 module containing the model to be optimized as
+    well as code to evaluate model performance.
+
+    ExperimentModule.default_parameters = {}
```

Comment:
as this is not compliant the OPF API, is there an industrial (really used) API for model parameter optimization? It'd be worth using that.

Reply:
There is not to my knowledge any standard interface for this.

This however is a nice and simple python API.

Reply:
https://www.khronos.org/nnef 
This is interesting, but seems like it's focused only at deep learning.